base/statistics.jl

```diff
@@ -467,7 +475,7 @@ end
 corzm(x::AbstractVector{T}) where {T} = one(real(T))
 function corzm(x::AbstractMatrix, vardim::Int=1)
     c = unscaled_covzm(x, vardim)
-    return cov2cor!(c, sqrt!(diag(c)))
+    return cov2cor!(c, collect(sqrt(c[i,i]) for i in 1:min(size(c)...)))
```

Comment:
I imagine `cov2cor!` returns `c`, so that the potential type change under this transformation should not matter much? If so, `collect` -> `Vector`? :)

Reply:
Does that work for a generator? I get:
`MethodError: no method matching Array{T,1} where T(::Base.Generator{UnitRange{Int64},getfield(Base, Symbol("##691#692")){Array{Float64,2}}})`

Reply:
Ah yes, not implemented yet :).